### DISCUSSION TITLE
docs: Improve documentation on how to handle reconections

### DISCUSSION THREAD

**Discussion by ljayatRabbit** [mikro-orm/mikro-orm#3675] [Q&A] [answered]
Hi!
I would like to know if there was a possibility to improve or detail how the developer can set behaviors for events like when the connection ends due to a network issue, database crash/restart, the database ends terminates the connection, etc.

These information could be useful to manage connection retrials, logging, and other things.

I tried to find how to catch those errors and also tried to find the correct way to get the client instance to make use of the driver api to handle this, but those paths didn't convince me.

If there is already something in the documentation about this topic, please dismiss this request, but it would be really helpful if you could provide the link to that part :)

**Comment by B4nan** [comment, marked as answer]
I am not sure if there is anything to document. Query execution and connection pooling is both handled by knex, I don't think you need to care about reconnection yourself anyhow, I believe failed query connections will be automatically removed from pool and new ones will be created if needed.

It's probably not very documented there either, but after some digging I found this PR https://github.com/knex/knex/pull/2017
